### PR TITLE
Added auth to DestinationModule

### PR DESCRIPTION
**Pull request by domaspoliakas**
As per title.

Any and all feedback welcome! I have a couple specific questions however:

1) Not sure how to approach testing. My gut feeling says that there's nothing to test, but if someone could confirm/deny this that'd I'd appreciate it.
2) Given that this is a breaking change that will require all destinations to be updated I was wondering how to approach it. Specifically I'm wondering how to approach this immediately post-merge since all destinations will be broken at that point. Would this affect other people, therefore I'd need to make sure all the destinations are updated asap? Or not? Any elaboration on this would be lovely

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,9 @@ docker/Dockerfiles/Hadoop/hadoop*.tar.gz
 /pubring.pgp.enc
 /scripts/readVersion
 /secring.pgp.enc
+
+# metals
+.bloop/
+.metals/
+.bsp/
+metals.sbt
```
Comment (nicflores) [inline .gitignore:38]: These can should probably go in your own ` ~/.gitignore_global`.

File: connector/src/main/scala/quasar/connector/destination/DestinationModule.scala [hunk 2]
```diff
@@ -33,6 +35,7 @@ trait DestinationModule {
 
   def destination[F[_]: ConcurrentEffect: ContextShift: MonadResourceErr: Timer](
       config: Json,
-      pushPull: PushmiPullyu[F])
+      pushPull: PushmiPullyu[F],
+      auth: UUID => F[Option[ExternalCredentials[F]]])
```
Comment (jsantos17) [inline connector/src/main/scala/quasar/connector/destination/DestinationModule.scala:39]: Would it make sense to make a type alias for this? Similar to `PushmiPullyu[F]`.

Comment (domaspoliakas) [reply connector/src/main/scala/quasar/connector/destination/DestinationModule.scala:39]: To be honest I was thinking the same thing after having to paste that all over the place. I'll get it done tomorrow morning 👍

**Comment by jsantos17** [comment]
On the topic of how to handle depending destinations:

I would implement the fixes in the datasources *before* merging this. You can publish an snapshot of this PR using `releaseSnapshot` in `sbt` so you can depend on it when fixing destinations. You can then open PRs on each datasource fixing this preemptively. That way you're not under pressure to fix the destinations.

After you have fixed all destination you would add a `breaking` tag to this PR and merge. After that's done `sbt-trickle` will create PR bumps in datasources. You can then merge the bump PRs into the fixes you previously implemented. That way nothing breaks.

I would be happy to help you along with this process if you wish.

**Comment by domaspoliakas** [comment]
Will clean up and pick this up on a separate branch

**Pull request closed without merging**